# Thinkful Drill #1: Describing Data

1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

5. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans. Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [41]:
import pandas as pd
import numpy as np

brady_bunch = pd.read_csv("Drill 1 Data.csv")
brady_bunch.head()

Name  Age        Magazine Title  % readers
0    Greg   14              TV Guide       20.0
1  Marcia   12  Entertainment Weekly       23.0
2   Peter   11     Pop Culture Today       17.0
3     Jan   10      SciPhi Phanatics        5.0
4   Bobby    8                 Other       35.0

In [42]:
brady_bunch.describe()

Age  % readers
count   7.000000   5.000000
mean    9.857143  20.000000
std     2.734262  10.816654
min     6.000000   5.000000
25%     8.000000  17.000000
50%    10.000000  20.000000
75%    11.500000  23.000000
max    14.000000  35.000000

#### 1) Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [43]:
import statistics
bb_mean = np.mean(brady_bunch["Age"])
bb_median = np.median(brady_bunch["Age"])
bb_mode = statistics.mode(brady_bunch["Age"])
bb_std = np.std(brady_bunch["Age"],ddof=1)
bb_stderr = bb_std / np.sqrt(len(brady_bunch["Age"]))
bb_variance = np.var(brady_bunch["Age"])

string_list = ["mean","median","mode","standard deviation","standard error","variance"]
values_list = [bb_mean,bb_median,bb_mode,bb_std,bb_stderr,bb_variance]
count = 0

for value in values_list:
        print("The " + string_list[count] + " value for brady bunch kids' ages is: " + str(value))
        count+=1

The mean value for brady bunch kids' ages is: 9.857142857142858
The median value for brady bunch kids' ages is: 10.0
The mode value for brady bunch kids' ages is: 8
The standard deviation value for brady bunch kids' ages is: 2.734262327610589
The standard error value for brady bunch kids' ages is: 1.03345401972
The variance value for brady bunch kids' ages is: 6.408163265306122


#### 2) Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

Choose mean for estimating central tendency because there are no major outliers in the data set to fluctuate the average. 
Choose standard deviation for estimating variance because it helps to visualize the difference between the data and central tendency (more so than standard error value and variance value).

#### 3) Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [44]:
print("Cindy's age before birthday: " + str(brady_bunch.loc[brady_bunch["Name"]=="Cindy","Age"]))
brady_bunch.loc[brady_bunch["Name"]=="Cindy","Age"] += 1
print("Cindy's age after birthday: " + str(brady_bunch.loc[brady_bunch["Name"]=="Cindy","Age"]))
bb_mean = np.mean(brady_bunch["Age"])
bb_median = np.median(brady_bunch["Age"])
bb_mode = statistics.mode(brady_bunch["Age"])
bb_std = np.std(brady_bunch["Age"],ddof=1)
bb_stderr = bb_std / np.sqrt(len(brady_bunch["Age"]))
bb_variance = np.var(brady_bunch["Age"])

string_list = ["mean","median","mode","standard deviation","standard error","variance"]
values_list = [bb_mean,bb_median,bb_mode,bb_std,bb_stderr,bb_variance]
count = 0

for value in values_list:
        print("The " + string_list[count] + " value for brady bunch kids' ages is: " + str(value))
        count+=1

Cindy's age before birthday: 5    6
Name: Age, dtype: int64
Cindy's age after birthday: 5    7
Name: Age, dtype: int64
The mean value for brady bunch kids' ages is: 10.0
The median value for brady bunch kids' ages is: 10.0
The mode value for brady bunch kids' ages is: 8
The standard deviation value for brady bunch kids' ages is: 2.516611478423583
The standard error value for brady bunch kids' ages is: 0.951189731211
The variance value for brady bunch kids' ages is: 5.428571428571429


Estimates that changed: mean, standard deviation, standard error, variance value

Estimates that did not change: median, mode

#### 4) Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

In [47]:
brady_bunch["Name"] = brady_bunch["Name"].str.replace("Oliver","Jessica")
brady_bunch.loc[brady_bunch["Name"] == "Jessica","Age"] = 1
print(brady_bunch[["Name","Age"]])
bb_mean = np.mean(brady_bunch["Age"])
bb_median = np.median(brady_bunch["Age"])
#bb_mode = statistics.mode(brady_bunch["Age"])
bb_std = np.std(brady_bunch["Age"],ddof=1)
bb_stderr = bb_std / np.sqrt(len(brady_bunch["Age"]))
bb_variance = np.var(brady_bunch["Age"])

# Multiple modes found, returning just the first mode
(values, counts) = np.unique(brady_bunch['Age'], return_counts=True)
ind = np.argmax(counts)
bb_mode = values[ind]


string_list = ["mean","median","mode","standard deviation","standard error","variance"]
values_list = [bb_mean,bb_median,bb_mode,bb_std,bb_stderr,bb_variance]
count = 0

for value in values_list:
        print("The " + string_list[count] + " value for brady bunch kids' ages is: " + str(value))
        count+=1

      Name  Age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    1
The mean value for brady bunch kids' ages is: 9.0
The median value for brady bunch kids' ages is: 10.0
The mode value for brady bunch kids' ages is: 1
The standard deviation value for brady bunch kids' ages is: 4.242640687119285
The standard error value for brady bunch kids' ages is: 1.60356745147
The variance value for brady bunch kids' ages is: 15.428571428571429


Based on the data, I would switch to median for central tendency because Jessica fluctuates the dataset values (outlier), and median is a bit more accurate measure than mode due to not having 1 specific mode for the dataset. I would switch to variance value for checking variance because it shows a larger variance between datasets which is expected after replacing Oliver(8) with Jessica(1). Could also stick with standard deviation (4.24 > 2.52, Jessica vs. Oliver).  

#### 5) On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans. Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [1]:
(20 + 23 + 17) / 3

20.0

Roughly 20% of adult Americans were Brady Bunch fans on the 50th anniversary of the show. Did not use SciPhi Phanatics data because it's specifically for scifi shows, not necessarily a good representation for fans of the Brady Bunch (non sci-fi film). Plus it's an outlier, so it would fluctuate the data, making the average a poor measurement for central tendency. 